Composer/plugins/azurePublish/src/node/azureProvisionUtil.ts

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { SubscriptionClient } from '@azure/arm-subscriptions';
+import { Subscription } from '@azure/arm-subscriptions/esm/models';
+import { ResourceManagementClient } from '@azure/arm-resources';
+import { ResourceGroup, GenericResource } from '@azure/arm-resources/esm/models';
+import { AzureBotService } from '@azure/arm-botservice';
+import { WebSiteManagementClient } from '@azure/arm-appservice';
+import { ResourceNameAvailability } from '@azure/arm-appservice/esm/models';
+import { CheckNameAvailabilityResponseBody } from '@azure/arm-botservice/esm/models';
+
+export enum AzureAPIStatus {
+    INFO = 'INFO',
+    PARAM_ERROR = 'PARAM_ERROR',
+    ERROR = "ERROR"
+}
+
+export enum AzureResourceProviderType {
+    QnA = 'Microsoft.CognitiveServices',
+    Luis = 'Microsoft.CognitiveServices',
+    CosmosDB = 'Microsoft.DocumentDB',
+    BlobStorage = 'Microsoft.Storage',
+    ApplicationInsights = 'Microsoft.Insights',
+    WebApp = 'Microsoft.Web',
+    Bot = 'Microsoft.BotService',
+}
+
+export class AzureProvisionUtil {
+
+    /**
+     * credentials from azure login
+     */
+    private credentials: any;
+
+    /**
+     * customized injected logger
+     */
+    private logger: any;
+
+    constructor(credentials: any) {
+        this.credentials = credentials;
+    }
+
+    /**
+     * GetUserSubscriptions based on the credentials of users
+     */
+    public async GetUserSubscriptions(): Promise<Array<Subscription>> {
```

Comment:
How is this different than the existing /azure/subscriptions API endpoint?  Do we need it?

Reply:
> How is this different than the existing /azure/subscriptions API endpoint? Do we need it?

These are different implementations, one is raw http call, the other is azure-js-sdk pakcage, I'm replacing the api call with sdk function call.